File: CleanAndroid/app/src/main/java/de/phillippbertram/cleanandroid/main/MainPresenter.java
package de.phillippbertram.cleanandroid.main;

import de.phillippbertram.cleanandroid.internal.mvp.RxPresenter;
import io.reactivex.Observable;
import java.util.concurrent.TimeUnit;

/**
 * @author [PBE] Phillipp Bertram | phillipp.bertram@lmis.de
 */

final class MainPresenter extends RxPresenter<MainView> {

    MainPresenter() {
        Observable.interval(1, TimeUnit.SECONDS).compose(bindToLifecycle()).subscribe();
    }
}

File: CleanAndroid/app/src/main/java/de/phillippbertram/cleanandroid/main/MainPresenter.kt
package de.phillippbertram.cleanandroid.main

import de.phillippbertram.cleanandroid.internal.mvp.RxPresenter
import io.reactivex.Observable
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable


/**
 * @author [PBE] Phillipp Bertram | phillipp.bertram@lmis.de
 */
class MainPresenter internal constructor() : RxPresenter<MainView>()
